presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PayloadJoinOptimizer.java

```diff
@@ -279,11 +337,24 @@ private PlanNode rewriteScanFilterProject(PlanNode planNode, RewriteContext<Join
                 return planNode;
             }
 
+            List<VariableReferenceExpression> outputCols = planNode.getOutputVariables();
             if (!ImmutableSet.copyOf(planNode.getOutputVariables()).containsAll(joinKeys)) {
```

Comment:
Not related to this PR, but here we can just check `planNode.getOutputVariables().containsAll(joinKeys)`. Don't need to create a ImmutableSet

Reply:
checking set containment may be more efficient than lists